src/charm.py

```diff
@@ -119,44 +127,48 @@ def legacy_v1_relations(self):
     def _on_tracing_request(self, e: RequestEvent):
         """Handle a remote requesting a tracing endpoint."""
         logger.debug(f"received tracing request from {e.relation.app}: {e.requested_receivers}")
-        self._update_tracing_relations()
+        self._update_tracing_v2_relations()
 
     def _on_tracing_relation_created(self, e: RelationEvent):
-        return self._update_tracing_relation(e.relation)
+        if not self._tracing.is_v2(e.relation):
+            self._publish_v1_data(e.relation)
+            # if this is the first legacy relation we get, we need to update ALL other relations
+            # as we might need to add all legacy protocols to the mix
+            self._update_tracing_v2_relations()
 
     def _on_tracing_relation_joined(self, e: RelationEvent):
-        return self._update_tracing_relation(e.relation)
+        if not self._tracing.is_v2(e.relation):
+            self._publish_v1_data(e.relation)
 
     def _on_tracing_relation_changed(self, e: RelationEvent):
-        return self._update_tracing_relation(e.relation)
+        if not self._tracing.is_v2(e.relation):
+            self._publish_v1_data(e.relation)
+
+    def _update_tracing_v1_relations(self):
+        for relation in self.model.relations[self._tracing._relation_name]:
+            if not self._tracing.is_v2(relation):
+                self._publish_v1_data(relation)
 
-    def _update_tracing_relation(self, relation: Relation):
+    def _publish_v1_data(self, relation: Relation):
         # we have a relation event; it might be caught by the v2 requirer
         # wrapper and turned into a `request` event, but also maybe not because
         # the remote isn't talking v2
-        if self._tracing.is_v2(relation):
-            # we will handle this as self._tracing.on.request on a different path
-            return
 
         if not self._is_legacy_v1_relation(relation):
-            logger.error(f"relation {relation} is not tracing v1 nor v2. Skipping...")
+            logger.error(f"relation {relation} is not tracing v1. Skipping...")
             return
 
         logger.debug(f"updating legacy v1 relation {relation}")
         # in v1, 'receiver' was called 'ingester'.
         receivers = [
-            tracing_v1.Ingester(protocol=p, port=self.tempo.receiver_ports[p])
+            tracing_v1.Ingester(protocol=p, port=self.tempo.receiver_ports[p], path=f"/{p}")
```

Comment:
Same here with `path` - I believe in the current idea it's supposed to be empty? I still think that it's good to have it in the protocol as it might become handy in future.